sbi/inference/abc/mcabc.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+from pyro.distributions.empirical import Empirical
+from torch.distributions.distribution import Distribution
+from sbi.inference.abc.abc_base import ABCBASE
+from typing import Callable, Optional, Union, Tuple
+
+import torch
+from torch import Tensor, ones
+from numpy import ndarray
+
+
+class MCABC(ABCBASE):
+    def __init__(
+        self,
+        simulator: Callable,
+        prior,
+        x_o: Union[Tensor, ndarray],
+        distance: Union[str, Callable] = "l2",
+        num_workers: int = 1,
+        simulation_batch_size: int = 1,
+        show_progress_bars: bool = True,
+    ):
+        """Monte-Carlo Approximate Bayesian Computation (Rejection ABC).
+
+        Args:
+
+        See docstring of `ABCBASE` class for all arguments.
+        """
+
+        super().__init__(
+            simulator=simulator,
+            prior=prior,
+            x_o=x_o,
+            distance=distance,
+            num_workers=num_workers,
+            simulation_batch_size=simulation_batch_size,
+            show_progress_bars=show_progress_bars,
+        )
+
+    def __call__(
+        self,
+        num_simulations: int,
+        eps: Optional[float] = None,
+        quantile: Optional[float] = None,
+        return_distances: bool = False,
+    ) -> Union[Distribution, Tuple[Distribution, Tensor]]:
+        r"""Run MCABC.
+        
+        Args:
+            num_simulations: Number of simulations to run.
+            eps: Acceptance threshold $\epsilon$ for distance between observed and
+                simulated data.
+            quantile: Upper quantile of smallest distances for which the corresponding
+                parameters are returned. Exactly one of quantile or `eps` have to be
+                passed.
```

Comment:
So quantile=0.99 will give the best 1% or quantile=0.01 will give the best 1%? Maybe add such an example to the docstring.